src/classes/RD2_ScheduleService.cls

```diff
@@ -535,6 +534,84 @@ public without sharing class RD2_ScheduleService {
         return schedulesByRdId;
     }
 
+    public List<ActiveSchedule> getVisualizedSchedules(npe03__Recurring_Donation__c rd) {
+        List<ActiveSchedule> activeSchedules = new List<ActiveSchedule>();
+
+        if (rd == null) {
+            return activeSchedules;
+        }
+
+        List<RecurringDonationSchedule__c> schedules = getExistingSchedules(rd);
+        if (schedules == null) {
+            return activeSchedules;
+        }
+
+        Boolean first1stAnd15thFound = false;
+        for (RecurringDonationSchedule__c schedule : schedules) {
+
+            // This block collapses 1st and 15th schedules into a single row and nulls out day of month
+            if (schedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH) {
+                if (first1stAnd15thFound) {
+                    first1stAnd15thFound = false;
+                    continue;
+                }
+                first1stAnd15thFound = true;
+                schedule.DayOfMonth__c = null;
+            }
+
+            ActiveSchedule activeSched = new ActiveSchedule(schedule);
+            if (schedule.StartDate__c <= currentDate) {
```

Comment:
Since getExistingSchedules only returns active schedules, it is impossible for that scenario to exist.  The only thing that actually happens here is separation of current from future.

Reply:
If we have time, it would be great to make it display data based on the backdated `currentDate` but not required.